File: server/src/entities/album-user.entity.ts
import { User } from 'src/database';
import { AlbumEntity } from 'src/entities/album.entity';
import { AlbumUserRole } from 'src/enum';

export class AlbumUserEntity {
  albumId!: string;
  userId!: string;
  album!: AlbumEntity;
  user!: User;
  role!: AlbumUserRole;
}

File: mobile/lib/infrastructure/entities/album_user.entity.dart
import 'package:drift/drift.dart';
import 'package:immich_mobile/domain/models/album_user.model.dart';
import 'package:immich_mobile/infrastructure/entities/user.entity.dart';
import 'package:immich_mobile/infrastructure/entities/remote_album.entity.dart';
import 'package:immich_mobile/infrastructure/utils/drift_default.mixin.dart';

class AlbumUserEntity extends Table with DriftDefaultsMixin {
  const AlbumUserEntity();

  TextColumn get albumId =>
      text().references(RemoteAlbumEntity, #id, onDelete: KeyAction.cascade)();

  TextColumn get userId =>
      text().references(UserEntity, #id, onDelete: KeyAction.cascade)();

  IntColumn get role => intEnum<AlbumUserRole>()();

  @override
  Set<Column> get primaryKey => {albumId, userId};
}
